azurelinuxagent/daemon/env.py

```diff
@@ -100,6 +107,33 @@ def handle_dhclient_restart(self):
            self.dhcp_handler.conf_routes()
            self.dhcpid = newpid
 
+    def handle_fstab_update(self):
+        """
+        Look for changes in fstab, mount if the fstab has new modified stamp 
+        If error, only re-check every minute to give an admin
+         time to correct it and avoid filling the log
+        """
+        fstabCurrentModifiedStamp = os.path.getmtime("/etc/fstab")
+        if fstabCurrentModifiedStamp != self.fstabModTime and \
+                            datetime.datetime.now() >  \
+                            self.lastNotice + datetime.timedelta(seconds=60):
+            ret, output = shellutil.run_get_output("mount -av")
```

Comment:
@hglkrijger it does have a side effect of mounting only the points umounted since the last (re)boot, which is something that's going to happen on the next reboot (which could happen at anytime) anyway. The main problem is we're trying so hard to prevent people to shoot themselves in the foot. As a result of this side effect, we're really going to annoy someone who's trying to umount something in the fstab by remounting it just after the fstab change. However this does not really seem even like a rare use case to me.

Based on Craig's comment from other PR, --fake does not really raise issues for most of the failure modes that could lead to boot issues.